phac-nml/irida-next
# Samples table metadata fields toggle

## What does this PR do and why?
_Describe in detail what your merge request does and why._

Updated both Project > Sample and Group > Samples pages to include a toggle that, when on, will display all metadata fields for the displayed samples.

Resolves #281

## Screenshots or screen recordings
_Screenshots are required for UI changes, and strongly recommended for all other pull requests._

### Metadata toggle
![image](https://github.com/phac-nml/irida-next/assets/11295750/34065fbb-873e-4689-8552-fea0248a3eae)

### Table with selected fields
![image](https://github.com/phac-nml/irida-next/assets/11295750/c24fdb12-7c66-45fa-9512-7341a86a1dc2)

## How to set up and validate locally
_Numbered steps to set up and validate the change are strongly suggested._

1. Ensure you re-seed the database as metadata fields have been added
1. Go to a project sample page (e.g. http://localhost:3000/yersinia/yersinia-pseudotuberculosis/outbreak-2023/-/samples)
1. Toggle the metadata, it should update the table with metadata

## PR acceptance checklist
This checklist encourages us to confirm any changes have been analyzed to reduce risks in quality, performance, reliability, security, and maintainability.
